src/commands/echo_command.c

```diff
@@ -0,0 +1,116 @@
+#include "commands.h"
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+#include <stdlib.h>
+#include <ctype.h> 
+#include "../../libft/libft.h"
+#include "../parser/parser.h"
+
+// possible strings
+// (            String with spaces to trim        )
+// (      "string with spaces to trim up to inverted commas"           )
+// (    string followed by pipe (|)    )
+// ("")
+// ( )
+// (		whitespace         4 dddaaaaaayyyyyys)
+// (	"	whitespace         4 dddaaaaaayyyyyys, with quotes      ")
+// (	"	quotes with \"\" quotes"      ")
+
+int	get_substr_len(const char *input)
+{
+	const int	strlen = ft_strlen(input);
+	int			i;
+
+	i = 0;
+	while (input[i])
+	{
+		if (input[i] == '|')
```

Comment:
Here we need to find out all the possible combinations with echo and make a function to check if the char is one of those for example | , <, <<, >, >>